workspaces/ui-v2/src/pages/docs/components/EndpointTOC.tsx

```diff
@@ -33,47 +34,58 @@ export function EndpointTOC(props: EndpointTOCProps) {
     <List dense>
       {props.query === null &&
       props.requests.length === 0 &&
-      props.responses.length === 0 ? (
+      Object.keys(props.responsesByStatusCode).length === 0 ? (
         <Typography className={classes.none}>No bodies documented.</Typography>
       ) : null}
 
       {props.query && (
         <EndpointTOCRow
           label={'Query Parameters'}
           anchorLink="query-parameters"
-          detail={<>consumes query string</>}
+          detail={<></>}
         />
       )}
 
-      {props.requests.map((request) => (
+      {props.requests.length > 0 && (
         <EndpointTOCRow
-          key={request.requestId}
           label={'Request Body'}
-          anchorLink={request.requestId}
+          anchorLink={'request-body'}
```

Comment:
Updated the anchor links - in a single endpoint page there are:
- 1 or 0 query parameters
- 1 or 0 request (with multiple / polymorphic request bodies)
- x number of status codes (with multiple request bodies)